### PR TITLE
DOC: special: add docstrings for various convenience functions

### PR DESCRIPTION
**Pull request by person142**
#### Reference issue
None

#### What does this implement/fix?
Add

- Parameters/Returns sections
- See Also sections
- Examples

for `radian`, `cosdg`, `sindg`, `tandg`, `cotdg`, `log1p`, `cosm1`,
and `round`.

#### Additional information
None

### CODE REVIEW

**Comment by person142** [comment]
CI failures appear to be unrelated.

**Review by tylerjereddy — APPROVED** [review]
Thanks Josh, these are big docstring improvements. Indeed, I learned about some functions  that could help for some of my computational geometry side projects.

CI fail unrelated, in it goes!

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -1628,23 +1628,111 @@ def add_newdoc(name, doc):
...
+
+    See Also
+    --------
+    sindg, tandg, cotdg
+
+    Examples
+    --------
+    >>> import scipy.special as sc
+
+    It is more accurate than using cosine directly.
```
Comment (pvanmulbregt) [inline scipy/special/add_newdocs.py]: `cosine` is ambiguous here, as SciPy doesn't have it's own `cosine` function. (Well it does have two `cosine` functions, in `scipy.signal.windows` and `scipy.spatial`, but neither of those is what you are referring to here, and `scipy.cos` is deprecated.)
Instead use `np.cos`.  E.g.

```
It is more accurate than using np.cos directly.
```
The same comment applies in a few places below --- I'll add a simplified comment at each location.

File: scipy/special/add_newdocs.py [hunk 2]
```diff
@@ -1628,23 +1628,111 @@ def add_newdoc(name, doc):
...
+
+    See Also
+    --------
+    sindg, cosdg, tandg
+
+    Examples
+    --------
+    >>> import scipy.special as sc
+
+    It is more accurate than using cotangent directly.
```
Comment (pvanmulbregt) [inline scipy/special/add_newdocs.py]: Perhaps
```
It is more accurate to use cotdg than to use np.tan to compute the cotangent directly.
```

File: scipy/special/add_newdocs.py [hunk 3]
```diff
@@ -6374,9 +6462,39 @@ def add_newdoc(name, doc):
...
+    See Also
+    --------
+    expm1, cosm1
+
+    Examples
+    --------
+    >>> import scipy.special as sc
+
+    It is more accurate than using ``log(1 + x)`` directly for ``x``
+    near 0. Note that in the below example ``1 + 1e-17 == 1`` to
```
Comment (pvanmulbregt) [inline scipy/special/add_newdocs.py]: I wonder if this is the most compelling example? It is true that the relative error is 1 for `x=1e-17`, but that might be discounted by the reader as `x` is smaller than `np.finfo(float).eps`.  For `x=1e-8` say, the relative error in using `np.log(1+x)` is `1e-10`, which is many orders of magnitude bigger than one might expect. Just a thought.

File: scipy/special/add_newdocs.py [hunk 4]
```diff
@@ -8307,9 +8484,38 @@ def add_newdoc(name, doc):
...
+
+    See Also
+    --------
+    cosdg, tandg, cotdg
+
+    Examples
+    --------
+    >>> import scipy.special as sc
+
+    It is more accurate than using sine directly.
```
Comment (pvanmulbregt) [inline scipy/special/add_newdocs.py]: sine -> np.sin

File: scipy/special/add_newdocs.py [hunk 5]
```diff
@@ -8583,9 +8789,38 @@ def add_newdoc(name, doc):
...
+
+    See Also
+    --------
+    sindg, cosdg, cotdg
+
+    Examples
+    --------
+    >>> import scipy.special as sc
+
+    It is more accurate than using tangent directly.
```
Comment (pvanmulbregt) [inline scipy/special/add_newdocs.py]: tangent -> np.tan

**Pull request merged**